test/persistence-hooks.suite.js

```diff
@@ -2946,6 +2946,252 @@ module.exports = function(dataSource, should, connectorCapabilities) {
 
     });
 
+    describe('PersistedModel.upsertWithWhere', function() {
+      var createdId = null;
+      var createdExtra = null;
+      var createdName = null;
+
+      it('triggers hooks in the correct order on create', function(done) {
+        monitorHookExecution();
+        TestModel.upsertWithWhere({ where: { extra: 'e0101' }},
+          { id: 'not-found', name: 'not found' },
+          function(err, record, created) {
+            if (err) return done(err);
+            createdId = record.id;
+            createdExtra = record.extra;
+            createdName = record.name;
+            hookMonitor.names.should.eql([
+              'access',
+              'before save',
+              'persist',
+              'loaded',
+              'after save',
+            ]);
+            done();
+          });
+      });
+
+      it('triggers hooks in the correct order on update', function(done) {
+        monitorHookExecution();
+        TestModel.upsertWithWhere({ where: { extra: 'e0101' }},
+          { name: 'new name' },
+          function(err, record, created) {
+            if (err) return done(err);
+            hookMonitor.names.should.eql([
+              'access',
+              'before save',
+              'persist',
+              'loaded',
+              'after save',
+            ]);
+            done();
+          });
+      });
```

Comment:
@Amir-61 
I tried if (dataSource.connector.upsertWithWhere) {
              hookMonitor.names.should.eql([
                ...
              ]);
            } else {
              hookMonitor.names.should.eql([
                ...
              ]);
            }
I observed , it never validates the else part. It takes [] and [ values].  Can you please point me to a test case where it is been implemented for reference.